src/spatiotemporalembedding.jl

```diff
@@ -208,31 +188,96 @@ function Base.show(io::IO, em::SpatioTemporalEmbedding{Φ,BC, X}) where {Φ,BC,X
     end
 end
 
+#####################################################################################
+#                                   CONSTRUCTORS                                    #
+#####################################################################################
+
+function SpatioTemporalEmbedding(
+		s::AbstractArray{<:AbstractArray{T,Φ}},
+		D, τ, B, k, boundary::BC
+		) where {T,Φ, BC<:AbstractBoundaryCondition}
+    if (BC <: ConstantBoundary) && typeof(boundary.c) != T
+	     throw(ArgumentError(
+		"Boundary value must be same element type as the timeseries data."))
+	end
+	X = (D+1)*(2B+1)^Φ
+	τs = Vector{Int}(undef,X)
+	βs = Vector{CartesianIndex{Φ}}(undef,X)
+	n = 1
+	for d=0:D, α = Iterators.product([-B*k:k:B*k for φ=1:Φ]...)
+		τs[n] = d*τ
+		βs[n] = CartesianIndex(α)
+		n +=1
+	end
+	return SpatioTemporalEmbedding{X}(τs, βs, boundary, size(s[1]))
+end
+
 
-"""
-	reconstruct(s::AbstractArray{<:AbstractArray{T,Φ}}, em)
-Reconstruct the spatial timeseries `s` represented by a `Vector` of `AbstractArray`
-states using the embedding struct `em` of type [`AbstractSpatialEmbedding`](@ref).
 
-Returns the reconstruction in the form of a [`Dataset`](@ref) where each row is a
-reconstructed state and they are ordered first through linear indexing into each state
-and then incrementing in time.
+function indices_within(radius, dimension)
+    #Return indices β within hypersphere with radius
+    #Floor radius to nearest integer. Does not lose points
+    r = floor(Int,radius)
+    #Hypercube of indices
+    hypercube = CartesianIndices((repeat([-r:r], dimension)...,))
+    #Select subset of hc which is in Hypersphere
+
+    βs = [ β for β ∈ hypercube if norm(β.I) <= radius ]
+end
+
 """
-function reconstruct(s::AbstractArray{<:AbstractArray{T,Φ}},
-	em::AbstractSpatialEmbedding{Φ,BC,X}
-	) where {T<:Number,Φ,BC,X}
-	timesteps = (length(s) - get_τmax(em))
-	num_pt    = get_num_pt(em)
-	L         = timesteps*num_pt
-
-	pt_in_space = CartesianIndices(s[1])
-	lin_idxs    = LinearIndices(s[1])
-	data = Vector{SVector{X,T}}(undef,L)
-	recv = zeros(T,X)
-	@inbounds for t in 1:timesteps, α in pt_in_space
-		n = (t-1)*num_pt+lin_idxs[α]
-		em(recv,s,t,α)
-		data[n] = recv
-	end
-	return Dataset(data)
+    LightConeEmbedding(s, N, τ, r₀, c, bc) → SpatioTemporalEmbedding
+Create a [`SpatioTemporalEmbedding`](@ref) struct that
+includes spatial and temporal neighbors of a point based on the notion of
+a _sphere of influence_.
+As an example, in a one-dimensional system with `N=2` timesteps, `τ=2`,
+`r₀ = 1` initial radius, and speed `c=1`
+the resulting embedding might look like:
+
+    __________o__________
+    _________xxx_________
+    _____________________
+    _______xxxxxxx_______
```

Comment:
This is one AMAZING example. Good job!

However, based on the example I think we have to rework this. I believe `r_o` should state the area around the point we are already, `o` (in your plot). `c` means the "amount of new points in each spatial direction" (so that c = 1 adds 1 point to the left _and_ one point to the right). I think this is the current idea as well?

I think `τ=2` is treated wrong as well. Shouldhn't first and second line also have an emptu line inbetween them? the timesteps `N` seem totally ok though.

Reply:
What do you mean?

An embedding is done on a point that exists. An embedding has nothing to do with a prediction. We use embeddings for predictions but this is irrelevant.

In standard delay coordinates given a point (that exists) you delay embed it using the point itself, as well as a past τ away, a past 2τ away etc.

In the current light cone case I would expect that you use the point itself, (optionally with a radius r0), then go τ into the past and expand your radius by r0 + τ*c. Include all points in this radius. Go into the past 2τ and repeat.

Reply:
If you want we can change the meaning of `r_0` to be the radius at the point that is to be predicted.

Reply:
I agree, an embedding is done on data that exists.
However the lightcone embedding is _highly_ motivated by prediction.

What you expect is exactly what is happening.

__________o_________
_________x . x________
____________________
_______xxxxxxx________
We are at `.` and want to predict `o`

Reply:
Alright. This is a fine example, please change it so that it has both the `.` and the `o` the it is understandable. 

I am not arguing much about how we do it, I am arguing about how easy it is to understand. You can say in this example that this embedding is motivated by prediction as well.

I would do the docstrings myself, but I want you to first do this change, because I am not yet confident that I know _exactly_ what each argument _exactly_ means.

Reply:
done already

Reply:
Ah its already changed lol. damn you github!